### PR TITLE
Add a new 'Getting Started' guide for Embedded Swift and writing an application for a microcontroller

### PR DESCRIPTION
**Pull request by kubamracek**
Adding a new "Getting Started" guide, alongside the other guides/tutorials. This one describes how to get started with Embedded Swift by building a simple firmware for a Raspberry Pi Pico, with the option of running the result in a simulator instead of a physical device.

### CODE REVIEW

File: getting-started/_use-cases.md [hunk 1]
```diff
@@ -55,6 +55,17 @@ Here are some examples of the many use cases of Swift, in case you want to jump
 
     <a href="/getting-started/swiftui" class="cta-secondary">Start tutorial</a>
   </li>
+
+  <li class="use-case">
+    <h3>Embedded Application for a Microcontroller</h3>
+    <p class="description">
+      Learn how to create an application in Embedded Swift will run on an embedded device or in a simulator.
```
Comment (cthielen) [inline getting-started/_use-cases.md:62]: "… **using** Embedded Swift **that** will run …"

File: getting-started/_use-cases.md [hunk 2]
```diff
@@ -55,6 +55,17 @@ Here are some examples of the many use cases of Swift, in case you want to jump
...
+      <br><br>
```
Comment (federicobucchi) [inline getting-started/_use-cases.md:63]: remove `<br><br>`

File: getting-started/_use-cases.md (hunk 1 above)
Comment (0xTim) [reply getting-started/_use-cases.md:62]: This is going to break the layout of the grid we have (which is deliberately 2x2). We could either add it to the "Go Further" section below or will need to discuss with the SWWG about how we want to tackle.

@kaishin @alexandersandberg any thoughts?

File: getting-started/embedded-swift/index.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+The Pico should then show up as a mounted volume in /Volumes (as RPI-RP2 in this case):
+
+```shell
+$ ls -al /Volumes
+lrwxr-xr-x   1 root  wheel     1B Jan  1 12:00 Macintosh HD@ -> /
+drwx------   1 kuba  staff    16K Dec 31  1969 RPI-RP2/
+```
+
+Simply copy the UF2 file to this volume:
```
[the remark this answers is not in the document]
Comment (rauhul) [reply getting-started/embedded-swift/index.md:158]: Yeah we should ditch "simply"; different things are simple for different folks, no need to prescribe difficulty in our guide

File: getting-started/_use-cases.md [hunk 4]
```diff
@@ -55,6 +55,17 @@ Here are some examples of the many use cases of Swift, in case you want to jump
 
     <a href="/getting-started/swiftui" class="cta-secondary">Start tutorial</a>
   </li>
+
+  <li class="use-case">
```
[the remark this answers is not in the document]
Comment (rauhul) [reply getting-started/_use-cases.md:59]: I dont think go further is the right place, that section is almost exclusively about language topics. The text even says: "Here are some hand-picked resources covering various Swift features."

[the remark this answers is not in the document]
Comment (federicobucchi) [reply getting-started/_use-cases.md:59]: What about in Using Swift and/or in Articles?

Screenshot 2024-06-03 at 12 41 37 PM

Screenshot 2024-06-03 at 12 45 13 PM

[the remark this answers is not in the document]
Comment (kubamracek) [reply getting-started/_use-cases.md:59]: Sounds good, added the reference to Documentation > Articles, too.

**Review by shahmishal — APPROVED** [review]

**Comment by shahmishal** [comment]
Do we have this resolved? https://github.com/apple/swift-org-website/pull/685#discussion_r1626605151

**Comment by shahmishal** [comment]
@swift-ci test

**Pull request merged**